### PR TITLE
Fix/follow up times

### PR DESCRIPTION
**Pull request by kirilklein**
## Summary by CodeRabbit

* **New Features**
  * Richer visualizations (effect-size, contingency, adjustment, follow-up coverage/duration/timelines, weight & gradient plots), encoding export/analysis, realistic outcome simulation, outcome/filtering plotters, optional encoding/weight visualizations in finetuning.

* **Bug Fixes**
  * Outcomes on follow-up boundaries included; group-wise follow-up durations prevented from going negative; stronger index-date validation.

* **Configuration**
  * New/extended plotting, finetune/uncensored/simulation presets and pagination controls.

* **Tests**
  * Expanded/updated tests for follow-ups, matching, outcomes, pipelines, uncensored validation, and XGBoost baselines.

* **Chores**
  * Centralized Azure-aware figure saving, CI/Windows pipeline updates, added dependency (umap-learn).

### CODE REVIEW

**Comment by ggideonebi-cmd** [comment]
Holden check

File: corebehrt/functional/utils/filter.py [hunk 1]
```diff
@@ -20,3 +21,76 @@
...
+        pass
+
+    # Strategy 3: Fallback to object dtype
+    return all_idx.astype("object"), exp_idx.astype("object")
+
+
+def _convert_to_int(idx: pd.Index) -> pd.Index:
+    """Convert index to int, raising exception if not possible."""
+    return pd.Index([int(x) for x in idx])
+
```
[the remark this answers is not in the document]
Comment (sanjivku79584) [reply corebehrt/functional/utils/filter.py:92]: Collect results
      ↓
Store period + number + colour + Big/Small
      ↓
Calculate statistics
      ↓
Display charts
      ↓
Generate a heuristic prediction

[the remark this answers is not in the document]
Comment (sanjivku79584) [reply corebehrt/functional/utils/filter.py]: Please send me give telegram colour pride app name jalaw game 🎯 give Yes
